### ISSUE TITLE
Cannot read property 'isConnected' of null

### ISSUE THREAD

**Issue by pierot** [phoenixframework/phoenix_live_view#737]
### Environment

* Elixir version (elixir -v): 1.10
* Phoenix version (mix deps): 1.4.16
* Phoenix LiveView version (mix deps): 0.10.0
* NodeJS version (node -v): 13.11.0
* NPM version (npm -v): 6.3.17
* Operating system: Linux
* Browsers you attempted to reproduce this bug on (the more the merrier): Chrome, Safari
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: Yes

### Actual behavior

Since the upgrade to 0.10.0 we regularly get the following error message in Sentry.

```
Cannot read property 'isConnected' of null
```

This happens here: https://github.com/phoenixframework/phoenix_live_view/blob/ec530aea4a2aac8a69f60d209e3a6f182de2d73c/assets/js/phoenix_live_view.js#L673

**Comment by josevalim** [comment]
Do you have a complete stacktrace? Or can you reproduce the error locally? Thanks!

**Comment by pierot** [comment]
I can reproduce it locally, yes.
We have a React app with a link that opens a modal like this:
```jsx
const stateOpts = canEdit ? {onClick: this.handleOpenModal} : {}
return (
  <a href="#" {...stateOpts} className="case-notes-show-trigger">
    {t("case.notes.trigger")} ({kase.notes.length})
  </a>
)
```
The moment the `#` appears in the URL, the error occurs. 

I shared the error from Sentry, https://sentry.io/share/issue/43fb58cd9bb5448ba649562f30800523/, but the info is very limited.

**Comment by josevalim** [comment]
Oh, fantastic! It may be a push state conflict. Is it possible for you to isolate the problem in a small app that you can share with us? It is going to make our lives much easier in fixing (and in guaranteeing we have the proper fix). Thanks!

**Comment by pierot** [comment]
While setting up a test project I followed an existing tutorial for adding `phoenix_live_view` to a new project and noticed the following:
```elixir
    "phoenix": "file:../deps/phoenix",
    "phoenix_html": "file:../deps/phoenix_html",
    "phoenix_live_view": "file:../deps/phoenix_live_view",
```
It then struck me that we have `phoenix_live_view` in our `package.json` like this: 
```elixir
    "phoenix_live_view": "^0.5.0"
``` 
.. so we are not using the latest version. When I upgraded the project where we saw the issue, the issue disappeared completely.

So, whoever stumbles upon this issue, keep the versions in sync or use the `file:...` syntax in your `package.json`.
